PIL/Image.py

```diff
@@ -1653,13 +1653,15 @@ def save(self, fp, format=None, **params):
            may have been created, and may contain partial data.
         """
 
+        filename = ""
         if isPath(fp):
             filename = fp
-        else:
-            if hasattr(fp, "name") and isPath(fp.name):
-                filename = fp.name
-            else:
-                filename = ""
+        elif sys.version_info >= (3, 4):
```

Comment:
I'm literally a year late on this, but this breaks support for passing in file objects (with a `name` property) in Python 3.4 and up. Previously the file name would be pulled from `fp.name` (assuming it was a valid path), but now because this check is in here for Python 3.4, it no longer pulls the file name. As a result, you must set the format explicitly whenever you are saving files now, as the `filename` will always be an empty string after this point (because a file object is not a path), which means it can't introspect the extension.

This is broken in Pillow 3.0.0 up to the latest master, and isn't documented anywhere.